functional_tests/README.md

```diff
@@ -0,0 +1,274 @@
+| HEADLESS   | Toggle headless browser mode, set to "False" to show the browser window                                                                                                        | True                           |
+| SLOW_MO    | Sets the Playwright slow mo mode in milliseconds                                                                                                                               | 0                              |
+| BROWSER    | Set the browser for playwright to use, must be one of `chromium`, `firefox`, or `webkit`.<br/> NOTE: Currently only chromium is supported and tests may fail in other browsers | chromium                       |
+| TRACE      | Toggle Playwright trace recording                                                                                                                                              | True                           |
+| TRACES_DIR | Sets the location to write Playwright trace files if TRACE is enabled                                                                                                          | <working_directory>/tmp_traces |
+
+## Viewing Failure Traces
+
+The tests record traces of all their actions, allowing you to follow through tests that previously ran and debug issues
+in remote environments.
+
+Our GitHub Action is configured to save traces of any failed scenario and upload them.
+
+### Viewing the Failure Trace from GitHub Actions
+
+1. Navigate to the failed Action functional test run, expand the "Upload Failure Traces" job step and click the download
+   link to download the zip file of all failed scenario traces.
+1. Unzip the downloaded file on the command line with
+    ```shell
+    unzip <path_to_file>
+    ```
+    (note that un-archiving using MacOS finder may not work as it recursively unzips the files inside, where we need the
+    files inside to remain zipped)
+1. This should leave you with a zip file for each failed scenario
+1. Open the traces zip files one at a time using the [Playwright Trace Viewer](https://playwright.dev/docs/trace-viewer)
+
+You should then be able to step through the failed tests and get a better idea of the state and cause of the failure.
+
+## Test Data Setup
+
+Some tests may require objects to be set up in the database, such as a user or set of pages that the feature relies
+upon. For this, we can use [Factory Boy](https://factoryboy.readthedocs.io/en/stable/orms.html#django) to see data
+directly into the database.
+
+### Importing Factories
+
+Factories that use Django models rely on Django being initialised. This causes issues if they are imported at a module
+level in the tests, as those imports then get run before Django has initialised, so can fail with a error like:
+
+```
+Models aren't loaded yet.
+```
+
+To work around this, the Factory classes must be imported within the step functions, so that
+
```

Comment:
`so that`? Unfinished sentence?

Reply:
This issue was resolved in a different way, so this will be removed entirely.